Canva/terraform-provider-sentry
# Migrate to Terraform SDK

## Intent
To migrate to Terraform's plugin SDK instead of depending on Terraform code-base itself. See: [Terraform SDK Announcement and Migration Steps](https://www.terraform.io/docs/extend/plugin-sdk.html)

### 🚨 IMPORTANT: Do not merge this PR yet. Before we need to migrate `canva/terraform-provider-buildkite` and update some bits in `canva/infrastructure` accordingly.

## Process & Testing
1. Find and replace of `github.com/hashicorp/terraform` `github.com/hashicorp/terraform-plugin-sdk`
2. Run `go mod tidy` to add, update and remove project dependencies.
3. Run `go build ./...` to confirm the project is still buildable.
4. Run `go test ./...` to confirm nothing break.
```bash
$ go test ./...
?   	github.com/canva/terraform-provider-sentry	[no test files]
ok  	github.com/canva/terraform-provider-sentry/sentry	(cached)
ok  	github.com/canva/terraform-provider-sentry/sentryclient	(cached)
```
5. Run `tf-sdk-migrator check` to verify that the project is now detected as successfully migrated. 
```bash
$ tf-sdk-migrator check
Go version 1.12.5: OK.
Checking whether provider uses Go modules...
Go modules in use: OK.
Checking version of github.com/hashicorp/terraform-plugin-sdk to determine if provider was already migrated...
Provider already migrated to SDK version 1.7.0
```
